openedx_events/content_authoring/signals.py

```diff
@@ -21,3 +21,26 @@
         "catalog_info": CourseCatalogData,
     }
 )
+
+
+# .. event_type: org.openedx.content_authoring.course.certificate_config.changed.v1
+# .. event_name: COURSE_CERTIFICATE_CONFIG_CHANGED
+# .. event_description: Fired when a course certificate configuration changes in Studio.
+# .. event_data: CertificateConfigData
+COURSE_CERTIFICATE_CONFIG_CHANGED = OpenEdxPublicSignal(
+    event_type="org.openedx.content_authoring.course.certificate_config.changed.v1",
+    data={
+        "certificate_config": CertificateConfigData,
+    }
+)
+
+# .. event_type: org.openedx.content_authoring.course.certificate_config.deleted.v1
+# .. event_name: COURSE_CERTIFICATE_CONFIG_DELETED
+# .. event_description: Fired when a course certificate configuration deletes in Studio.
+# .. event_data: CertificateConfigData
+COURSE_CERTIFICATE_CONFIG_DELETED = OpenEdxPublicSignal(
+    event_type="org.openedx.content_authoring.course.certificate_config.deleted.v1",
+    data={
+        "certificate_config": CertificateConfigData,
```

Comment:
Why do we need to pass all this information if it's been deleted? Can we just pass the `course_key` instead?

Reply:
I think we should. Why did I do this? It's all about the low coupling of the two services Studio and Credentials. We have to support it and not depend on each other. How can we efficiently delete resources in general? How do we do it with a regular http request in monolithic systems, i.e. explicitly pass an entity identifier, which is usually the primary key in a database table, or pass some unique field at the database level. Then the removal will be accurate and correct.

It’s not about our case now. Studio does not store the ID and does not know about the model from the Credentials database. It's good. course_key may or may not be a unique field. Studio doesn't know this for sure and shouldn't know. Only Credentials knows this. What can we do then when we keep services loosely coupled? Pass on as much meaningful information as possible and ask for Credentials to be removed. And Credentials itself will decide which record in the table needs to be deleted, since it is an information expert here. Which is what we do.

Reply:
> course_key may or may not be a unique field.

But there's only ever one credentials configuration for a given course_key, right? When would we ever have more than one?

Reply:
Not certainly in that way. Look at this model please https://github.com/openedx/credentials/blob/master/credentials/apps/credentials/models.py#L194-L230
Here course_id is just a CharField and not unique. And here is our constraint `unique_together = (("course_id", "certificate_type", "site")),)` which explains the behavior.

And in any case, it's not so important. After all, as I wrote above, Studio is not an information expert for Credentials and does not know the exact URI of this resource. It is important for us to maintain low coupling and sending more data will not be superfluous here.

Reply:
I see. I misunderstood in that I thought CertificateConfigData was the representation of all certificates for a given course (i.e. a cleaner version of the entire `certificates` attribute off of the CourseBlock), rather than being the configuration for one particular certificate type for a course.